tools/please_pex/pex_main.py

```diff
@@ -158,12 +158,13 @@ def get_code(self, fullname):
 
 
 def pex_basepath(temp=False):
-    if temp:
+    if temp or not os.access('~/.cache/pex', os.W_OK):
```

Comment:
same comment as other PR, this will return `False` if `~/.cache/pex` doesn't exist yet; it may be better to try to extract to $HOME and fall back if that fails.

(It would also need an `expanduser` as well, it doesn't do that implicitly)